File: src/script-jl/0009-ispalindrom.jl
# File Name: 0009-ISPALINDROM.JL
# Author: Qiangua
# Email: caicailiushui@protonmail.com
# Copyright © 2021, Qiangua, all rights reserved.
# Created: 22 November 2021


function ispalindrome(num::Integer)::Bool
    if num < 0 || (div(num,10) == 0 && num != 0)
        return false
    end

    revertedNum = 0;
    while num > revertedNum
        revertedNum = revertedNum *10 + mod(num,10)
        num = div(num,10)
    end

    return num == revertedNum || num == div(revertedNum,10)

end

using Test
@testset begin
    @test ispalindrome(121) == 1
    @test ispalindrome(-121) == 0
end

File: src/script-py/0009-ispalindrom.py
# File Name: 0009-ISPALINDROM.PY
# Author: Qiangua
# Email: caicailiushui@protonmail.com
# Copyright © 2021, Qiangua, all rights reserved.
# Created: 23 November 2021
#  -*- coding: utf-8 -*-

class Solution:
    def ispalindrome(self,num : int) -> bool:
        if num < 0 or (num % 10 == 0 and num != 0):
            return False
        revertedNum = 0
        while num > revertedNum:
            revertedNum = revertedNum * 10 + revertedNum % 10
            num = num / 10

        return num == revertedNum or num == revertedNum/10



import unittest
class TestLeetcode(unittest.TestCase):
    def test1(self):
        num = 121
        sol = Solution()
        expect = True
        actual = sol.ispalindrome(num)
        print(actual)
        self.assertEqual(expect,actual)


if __name__ == "__main__":
    unittest.main()